src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/DeliverymenRequestResolver.java

```diff
@@ -203,4 +201,64 @@ public MutableComponent getRequesterDisplayName(
     {
         return Component.translatable(TranslationConstants.COM_MINECOLONIES_COREMOD_JOB_DELIVERYMAN);
     }
+
+    /**
+     * Try to rebalance the queues of current dman requests linked to this warehouse
+     */
+    public void rebalanceCurrentRequests(@NotNull final IRequestManager manager)
+    {
+        // ideally we'd do a little work each tick to spread out the load, but sadly the colony tick only happens every ~25s
+        final Set<IRequest<?>> alreadyMoved = new HashSet<>();
+        final List<ICitizenData> workers = getResolveAbleDeliverymen(manager);
+        for (final ICitizenData worker : workers)
+        {
+            final JobDeliveryman workerJob = (JobDeliveryman) worker.getJob();
+            final List<IToken<?>> taskQueue = workerJob.getTaskQueue();     // assumed immutable copy, not live list
+            final boolean isWorking = worker.isWorking() && !worker.isPaused();
+
+            // don't attempt to reassign the current task unless the worker isn't actually working
+            for (int requestIndex = isWorking ? 1 : 0; requestIndex < taskQueue.size(); ++requestIndex)
```

Comment:
This is a problem with the courier prefetching (courier might be handling multiple request deliveries at the same time).

Reply:
That's why it has the condition to only move tasks that don't have matching source+destination pairs.  This also avoids splitting up shareable tasks even if they're not currently active.

Reply:
Ah okay, Ive been thinking a bit more about this and thought:
- Not rebalance ALL courier tasks but rather only rebalance if there are gross queue size differences
and Only rebalance a queue when more than x items (or x complexity)
- When we rebalance, not rebalance all at the same time but rather something like the bottom 10% 

- When triggering a pickup, actually do an estimate on how much has to be picked up and give a pickup task a "complexity" number which we then, when comparing queue sizes on assignment can use to balance tasks better.

Reply:
idk, I think that might be overcomplicating it.  It's very simple atm, and self-adjusting without any need to calculate complexities manually.

Reply:
The complexities would then be task in general, and we would use them for the actual normal balancing already to reduce the need to rebelance so much